lib/protobuf/field/field_hash.rb

```diff
@@ -0,0 +1,97 @@
+module Protobuf
+  module Field
+    class FieldHash < Hash
+
+      ##
+      # Attributes
+      #
+
+      attr_reader :field, :key_field, :value_field
+
+      ##
+      # Constructor
+      #
+
+      def initialize(field)
+        @field = field
+        @key_field = field.type_class.get_field(:key)
+        @value_field = field.type_class.get_field(:value)
+      end
+
+      ##
+      # Public Instance Methods
+      #
+
+      def []=(key, val)
+        super(normalize_key(key), normalize_val(val)) unless val.nil?
+      end
+
+      def store(key, val)
```

Comment:
Sure, will do. If you look hard, you'll see this is largely copy+pasta from `field_array.ruby` which doesn't try to be very DRY for things like (so I wasn't putting effort into improving on it... me = lazy programmer).